### PR TITLE
Request integration of Modules Taken Before Internship feature

### PR DESCRIPTION
**Pull request by tgqiang**
Changes implemented in this branch:

1. Back-end for this feature implemented by @nlzz22 
2. Front-end for this feature is implemented
3. List of UI changes:
  a) Added hyperlinking to module codes for pages for Non-Overlapping Modules, Modules Taken Prior To Others
  b) Renaming, reordering for home page and sidebar links
  c) Standardizing navigation options section for Module Information, Fixed and Tentative Module Mounting pages

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -40,7 +40,8 @@
     'components.handlers.students_affected_by_module.StudentsAffectedByModule',
     '/addModule', 'components.handlers.add_module_handler.AddModule',
     '/moduleTakenPriorToOthers', 'components.handlers.module_taken_prior_to_others.TakePriorTo',
-    '/nonOverlappingModules', 'components.handlers.non_overlapping_modules.NonOverlappingModules'
+    '/nonOverlappingModules', 'components.handlers.non_overlapping_modules.NonOverlappingModules',
+    '/moduleTakenPriorToInternship', 'components.handlers.modules_taken_prior_to_internship.TakePriorInternship'
```
Comment (helloqx) [inline app.py]: pylint: Line too long

File: components/handlers/modules_taken_prior_to_internship.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    def POST(self):
+        '''
+            Invoked when user wants to search for modules
+            before internship with a specified AY-Sem
+        '''
+        # will have input data as function is called from button
+        input_data = web.input()
+        ay_sem = input_data.sem
+        raise web.seeother('/moduleTakenPriorToInternship?sem=' + ay_sem)
```
Comment (helloqx) [inline components/handlers/modules_taken_prior_to_internship.py]: pylint: Missing final newline

File: test/test_app.py [hunk 3]
```diff
@@ -136,18 +136,32 @@ def test_goto_add_modules_page(self):
...
+
+
     def test_goto_non_overlapping_mods(self):
         '''
             Test if user can access non overlapping modules without request errors
         '''
         root = self.test_app.get('/')
+
         response = root.click(linkid="home-page", href="/nonOverlappingModules")
+
```
Comment (helloqx) [inline test/test_app.py]: pylint: Trailing whitespace

File: templates/modulesTakenPriorToInternship.html [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                </div>
+                <div class="form-group">
+                    <input type="submit"  value="Submit" class="btn btn-primary">
+                </div>
+            </form>
+        </div>
+    </div>
+	<br>
+
+	<table class="table display dataTable table-bordered table-hover text-center" id="modules-taken-prior-intern-table">
```
Comment (helloqx) [inline templates/modulesTakenPriorToInternship.html]: Should table be sortable?

Comment (helloqx) [inline templates/modulesTakenPriorToInternship.html]: Also, when no data, vs when data exists, column sizes vary greatly:
image

File: templates/base.html [hunk 5]
```diff
@@ -76,12 +76,15 @@
 				<h3><span class="glyphicon glyphicon-random"></span>&nbsp;&nbsp;&nbsp;Module Relations</h3>
 				<ul>
 					<li><a id="sidebar" href="/overlappingModules">Overlapping Modules</a></li>
-					<li><a id="sidebar" href="/moduleTakenPriorToOthers">Modules Taken Prior to Other Modules</a></li>
-					<li><a id="sidebar" href="/nonOverlappingModules">Non-Overlapping Modules</a></li>
-				    <!--ul>
-					   <li><a href="#">Before Specified Module</a></li>
-				        <li><a href="#">Before Internship</a></li>
-				    </ul-->
+					<li><a href="javascript:void(0)">Prior Modules Taken</a></li>
```
Comment (helloqx) [inline templates/base.html]: Useless link that closes sidebar, consider disabling

File: app.py (hunk 1 above)
Comment (tgqiang) [inline app.py]: I think this pylint issue should be ignored to make reading and modifying of the routes more easily.

File: templates/base.html (hunk 5 above)
Comment (tgqiang) [inline templates/base.html]: Sidebar will be redesigned at one of the refinement sprints. Please raise an issue in Github for this UI issue.

Comment (helloqx) [inline templates/base.html]: #38

File: templates/modulesTakenPriorToInternship.html (hunk 4 above)
Comment (helloqx) [inline templates/modulesTakenPriorToInternship.html]: Yeah, right now the table does not have sort/search, unlike the other pages

Comment (tgqiang) [inline templates/modulesTakenPriorToInternship.html]: image

Really? The sortings exist on my side.

Comment (helloqx) [inline templates/modulesTakenPriorToInternship.html]: 😱 appeared after resetting again
Resolved

**Pull request merged**